### ISSUE TITLE
Add suggestion to default `errorMessage` on how to fix missing/wrong license headers

### ISSUE THREAD

**Issue by scordio** [mathieucarbou/license-maven-plugin#531]
**Describe the feature**

Currently, I define a `check` execution in my projects to make sure each file has the proper license header and I let the CI fail if something is missing. This gives immediate feedback in PRs but the default error message doesn't give indications on how to fix the problem.

Without guidance, it's assumed that the PR author is or needs to get familiar with the `license-maven-plugin`, which is not always the case, especially for one-time contributors.

The project-specific contributing guidelines could contain some details about the license headers and how to fix the missing ones but I believe it would be more effective to have guidance directly in the error message. For this reason, I propose to define the default error message like the following:

```
Some files do not have the expected license header. Run license:format to update them.
```

I'm happy to raise a PR if the idea is accepted.

**Any alternative you've considered ?**

To give direct feedback to users, I currently define a custom error message with the content above:

```xml
<configuration>
  <errorMessage>Some files do not have the expected license header. Run license:format to update them.</errorMessage>
</configuration>
```

**Comment by hazendaz** [comment]
Go for it.  I do something very similar with maven wrapper with enforcer plugin.  Far too many IMO don't even read how it says to fix but it's there and having such a default gives us a better point to tell user to read the output.

I don't use check here myself so I don't recall our current message.  If you can please call it out here to help this conversation along.  Look forward to your PR.

Sent from my Verizon, Samsung Galaxy smartphone
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
From: Stefano Cordio ***@***.***>
Sent: Saturday, April 1, 2023 11:40:19 AM
To: mathieucarbou/license-maven-plugin ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [mathieucarbou/license-maven-plugin] Add suggestion to default `errorMessage` on how to fix missing/wrong license headers (Issue #531)


Describe the feature

Currently, I define a check execution in my projects to make sure each file has the proper license header and I let the CI fail if something is missing. This gives immediate feedback in PRs but the default error message doesn't give indications on how to fix the problem.

Without guidance, it's assumed that the PR author is or needs to get familiar with the license-maven-plugin, which is not always the case, especially for one-time contributors.

The project-specific contributing guidelines could contain some details about the license headers and how to fix the missing ones but I believe it would be more effective to have guidance directly in the error message. For this reason, I propose to define the default error message like the following:

Some files do not have the expected license header. Run license:format to update them.


I'm happy to raise a PR if the idea is accepted.

Any alternative you've considered ?

To give direct feedback to users, I currently define a custom error message with the content above:

<configuration>
  <errorMessage>Some files do not have the expected license header. Run license:format to update them.</errorMessage>
</configuration>

**Comment by scordio** [comment]
Sure, here it is:

```
Some files do not have the expected license header
```

Taken from: https://github.com/mathieucarbou/license-maven-plugin/blob/fdd8226c2e23571fefb1e556f7a52c39d386acb4/license-maven-plugin/src/main/java/com/mycila/maven/plugin/license/LicenseCheckMojo.java#L47-L48
